app/frontend/src/components/forms/submission/ManageSubmissionUsers.vue

```diff
@@ -0,0 +1,303 @@
+                      >
+                        <v-icon>remove_circle</v-icon>
+                      </v-btn>
+                    </td>
+                  </tr>
+                </tbody>
+              </template>
+            </v-simple-table>
+          </v-skeleton-loader>
+        </v-card-text>
+
+        <v-card-actions class="justify-center">
+          <v-btn class="mb-5 close-dlg" color="primary" @click="dialog = false">
+            <span>Close</span>
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+
+      <BaseDialog
+        v-model="showDeleteDialog"
+        type="CONTINUE"
+        @close-dialog="showDeleteDialog = false"
+        @continue-dialog="modifyPermissions(userToDelete.id, []); showDeleteDialog = false"
+      >
+        <template #title>Remove {{ userToDelete.username }}</template>
+        <template #text>
+          Are you sure you wish to remove
+          <strong>{{ userToDelete.username }}</strong
+          >? They will no longer have permissions for this submission.
+        </template>
+        <template #button-text-continue>
+          <span>Remove</span>
+        </template>
+      </BaseDialog>
+    </v-dialog>
+  </span>
+</template>
+
+<script>
+import { mapActions } from 'vuex';
+
+import { FormPermissions } from '@/utils/constants';
+import { rbacService, userService } from '@/services';
+
+export default {
+  name: 'ManageSubmissionUsers',
+  props: {
+    isDraft: {
+      type: Boolean,
+      required: true,
+    },
+    submissionId: {
+      type: String,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      dialog: false,
+      isLoadingTable: true,
+      showDeleteDialog: false,
+      userTableList: [],
+      userToDelete: {},
+
+      // search box
+      findUsers: null,
+      isLoadingDropdown: false,
+      userSearchResults: [],
+      userSearchSelection: null,
+    };
+  },
+  methods: {
+    ...mapActions('notifications', ['addNotification']),
+    // show users in dropdown that have a text match on multiple properties
+    async addUser() {
+      if (this.userSearchSelection) {
+        const id = this.userSearchSelection.id;
+        if (this.userTableList.some((u) => u.id === id)) {
+          this.addNotification({
+            type: 'warning',
+            message: `User ${this.userSearchSelection.username} is already in the list of team members.`,
+          });
+        } else {
+          this.modifyPermissions(id, [
+            FormPermissions.SUBMISSION_UPDATE,
+            FormPermissions.SUBMISSION_READ,
+          ]);
+        }
+      }
+      // reset search field
+      this.userSearchSelection = null;
+    },
+    filterObject(item, queryText) {
+      // eslint-disable-next-line
+      for (const [key, value] of Object.entries(item)) {
+        if (
+          value !== null &&
+          value.toLocaleLowerCase().includes(queryText.toLocaleLowerCase())
+        ) {
+          return true;
+        }
+      }
+    },
```

Comment:
For this predicate function, only the value is being used here - so you can leverage `Object.values()` instead and iterate over that.
```js
filterObject(item, queryText) {
  return Object.values(item).some((v) => v !== null && v.toLocaleLowerCase().includes(queryText.toLocaleLowerCase()))
},
```

Reply:
Yeah nice that works